# XBinary-Classifier

Here you can test our package using the example data set we have provided you with.
To start with make sure you have `pip install`-ed it and then start by importing it:

In [1]:
import xbinary_classifier

Next import the nesecary packages you might need to look at the data file:

In [2]:
from astropy.io import fits
from astropy.table import Table, vstack
import os
import pandas as pd
import numpy as np

In [3]:
home = os.path.expanduser('~')
sourcefile = home # + "/your-path-to-exampledataset.fits"

Now lets open the data file:

In [7]:
with fits.open(sourcefile) as hdul:
    table_hdu = hdul[1]
    df = pd.DataFrame(table_hdu.data)
df

NAME         RA       DEC   VMAG  BV_COLOR     PORB    FLUX  \
0    SAX J1603.9-7753  240.97500 -77.88500   0.00      0.00   0.0000   160.0   
1         2A 0521-720   80.11683 -71.96481  18.00      0.00   0.3400     9.0   
2     RX J0532.7-6926   83.17833 -69.43833   0.00      0.00   0.0000     0.5   
3          4U 1254-69  194.40500 -69.28911  19.10      0.30   0.1638    25.0   
4         GS 1124-684  171.61125 -68.67572  13.30      0.30   0.4325  3000.0   
..                ...        ...       ...    ...       ...      ...     ...   
296   IGR J00370+6122    9.29167  61.35972   9.65      0.56  15.6650     8.0   
297       4U 0115+634   19.63292  63.74000  14.50      1.40  24.3000     2.0   
298       2S 0114+650   19.51125  65.29167  11.00      1.20  11.6000     4.0   
299   IGR J01363+6610   23.95833  66.21111  13.29      1.39   0.0000     9.0   
300   IGR J01583+6713   29.57683  67.22319   0.00      0.00   0.0000    14.0   

     FLUX_MAX        LII       BII  VMAG_MIN  UB_COLOR  PULSE_PERIOD  target  
0         0.0  312.42847 -18.73093       0.0       0.0          0.00       0  
1        44.0  283.09949 -32.69058      19.0      -0.8          0.00       0  
2         0.0  279.93933 -32.15468       0.0       0.0          0.00       0  
3         0.0  303.48189  -6.42396       0.0       0.0          0.00       0  
4         0.0  295.30054  -7.07259       0.0       0.0          0.00       0  
..        ...        ...       ...       ...       ...           ...     ...  
296       0.0  121.22213  -1.46464       0.0       0.0          0.00       1  
297     350.0  125.92366   1.02574      16.3       0.3          3.61       1  
298       0.0  125.70998   2.56353       0.0       0.1      10008.00       1  
299       0.0  127.39482   3.72480       0.0       0.0          0.00       1  
300       0.0  129.35216   5.18871       0.0       0.0          0.00       1  

[301 rows x 14 columns]

Above we can see 301 X-ray Binary sources. The low mass binaries are marked with "0" at the `target` column, while the high mass ones are marked with "1".

Let us run our classifier just by giving it information about RA, DEC, VMAG, BV_COLOR, PORB, FLUX, FLUX_MAX, LII, BII, VMAG_MIN, UB_COLOR, PULSE_PERIOD about only one source of our choice.

We choose `SAX J1603.9-7753` as the example. From the table above we can see that it is a low mass X-ray binary, however we will not give that information to our code.

In [8]:
columns_to_select = ['RA', 'DEC', 'VMAG', 'BV_COLOR', 'PORB', 'FLUX', 'FLUX_MAX', 'LII', 'BII', 'VMAG_MIN', 'UB_COLOR', 'PULSE_PERIOD']

In [9]:
for column in df.columns:
    if df[column].dtype.byteorder == '>':  # Big-endian
        df[column] = df[column].values.byteswap().newbyteorder()

In [10]:
new_df = df.loc[df.index == 0, columns_to_select]

Let's check out the data sample we will give to our classifier:

In [12]:
print(new_df)

        RA     DEC  VMAG  BV_COLOR  PORB   FLUX  FLUX_MAX        LII  \
0  240.975 -77.885   0.0       0.0   0.0  160.0       0.0  312.42847   

        BII  VMAG_MIN  UB_COLOR  PULSE_PERIOD  
0 -18.73093       0.0       0.0           0.0  


In [30]:
values = new_df.iloc[0].tolist()

Below is the only line of code the user needs to run to check:

In [34]:
xbinary_classifier(values)

The X-ray Binary source has been classified as low mass


We can see that it has identified it correctly! 🥳